### DISCUSSION TITLE
2.18.1-1

### DISCUSSION THREAD

**Discussion by bb-qq** [bb-qq/r8152#404] [Announcements]
⚠️If you want to report a problem or get support, **please fill out the template and create a ticket in [Issues](https://github.com/bb-qq/r8152/issues)** instead of posting in the forum.

* The base code has been updated to version 2.18.1 which includes the following changes:
  * 🥳**Added support for new chip RTL8157 (5Gbps)** 🤩
    * However, the author has not been able to obtain a product with this chip, so it has not been confirmed that it works.
  * Comply with the kernel's new network driver interface, and many refactorings.
  * Many error handling improvements and additional workarounds, possibly stability improvements
  * ..And microcode updates
* Drivers are no longer released for the following platforms for DSM6.2/7.1 since Realtek has discontinued support for older kernels:
  *  evansport, comcerto2k, armadaxp, armada370, armada375
* Confirming stability is still in progress.

Note:

* The installation will fail the first time. After that, run the following command from the SSH terminal:
   `sudo install -m 4755 -o root -D /var/packages/r8152/target/r8152/spk_su /opt/sbin/spk_su`
* Selecting a proper version is needed depending on your DSM version.
  * DSM 7.2 and above
    * use packages with the suffix `_7.2`.
  * DSM 7.0 and DSM 7.1
    * use packages with the suffix `_7.1`.
  * DSM 6.2.x
    * use packages with the suffix `_6.2`.

<hr /><em>This discussion was created from the release <a href='https://github.com/bb-qq/r8152/releases/tag/2.18.1-1'>2.18.1-1</a>.</em>

**Comment by UNIPART82** [comment]
![KakaoTalk_20240623_165000205](https://github.com/bb-qq/r8152/assets/162960224/96b48267-f11d-4f8c-8db5-9a112b7fce46)
![KakaoTalk_20240623_165000205_01](https://github.com/bb-qq/r8152/assets/162960224/409bb9c1-71a7-434f-b1a1-6e508816fac7)
![KakaoTalk_20240623_165000205_02](https://github.com/bb-qq/r8152/assets/162960224/9b2d1317-5a46-412e-97ea-0ff09ae954b4)
![Test-Synology-NAS](https://github.com/bb-qq/r8152/assets/162960224/9d80d289-b62e-48c1-aa3e-fdcae9011f1a)



Xpenology : RR 24.6.6 ( SA6400 / DSM 7.2 )


┌ M/B : Asrock B250M Pro4 [https://www.asrock.com/mb/Intel/B250M%20Pro4/](url)
│
├ Intel I219V 1GB LAN = Default
│
└ USB 3.0 2.5GB LAN * 4EA [https://aliexpress.com/item/1005006520068954.html](url)


I have connected four 2.5GB LAN ports to the rear USB 3.0.


Only one out of the four is recognized.


Is there a way to recognize and use all four?


I am aware that there will be a bottleneck with the USB 3.0 interface.


후면 USB3.0에 4개의 2.5GB LAN을 연결했습니다.

4개 중 1개만 인식 하는데, 4개 모두 인식해 사용할 수 있는 방법은 없나요?

USB3.0 인터페이스 병목현상이 생길 것을 알고 있습니다.

**Reply by bb-qq** [reply]
Did you check this section?
https://github.com/bb-qq/r8152/wiki/Troubleshooting#multiple-identical-devices-do-not-work

**Reply by UNIPART82** [reply]
Thank you for the comment.

I have read the content several times, but I could not understand it.

If it's not too much trouble, I'm curious about whether it's possible or not and for what reason you asked to confirm the content.

I keep searching and searching...

I received a comment from another community stating that when using the same USB 2.5GB LAN, only one is recognized, and they are using a product with a different chip.

댓글 감사합니다.

내용을 여러 번 읽었습니다만, 이해 못 했습니다.

실례가 아니라면, 가능? 불가능? 어떤 이유로 내용확인을 물으셨는지 궁금합니다.

계속 검색 또 검색 . . .

다른 커뮤니티에서 같은 USB 2.5GB LAN 사용하면 하나만 인식해 Chip 다른 제품을 사용해 사용 중이라는 댓글을 받았습니다.

**Reply by TheS1R** [reply]
I just received my **WisdPi USB 5 Gbps Ethernet Adapter** (model WP-UT5) with Realtek RTL8157 chipset. The adapter connects at 1 Gbps. Any thoughts?
![20240808_130635](https://github.com/user-attachments/assets/471b2358-36de-4743-829e-3107b7435720)

**Reply by TheS1R** [reply]
I figured it out by rerouting connections through different switch and adjusting adapter configuration in DSM. I'm now successfully connected with 5 Gbps link to TP-Link TL-SX105 (_5 port 10Gbps unmanaged switch_). I am seeing about 3.35 Gbps using iperf3 — the limitation appears to be the USB 3.0 ports on the DS718+, which are spec'd at 4.8 Gbps (_theoretical_).

`# lsusb
  |__2-3         0bda:8157:3000 00  3.20 5000MBit/s 544mA 1IF  (WisdPi USB 5G Ethernet 000334C8D6B10110)
`

**Comment by WolfAGR** [comment]
Hi BB-qq,
Is there newer version of r8152-armadaxp-2.17.1-2_7.1.spk?
Cheers,
Wolfairga

**Reply by bb-qq** [reply]
Unfortunately, new drivers for armadaxp will be no longer released because Realtek has discontinued support for older kernels.

**Reply by WolfAGR** [reply]
Thank you for the explanation.
